File: Python/decorator_pattern/cafe_adicional_de_acucar.py
from cafe_decorator import CafeDecorator

class CafeAdicionalDeAcucar(CafeDecorator):
    def __init__(self, cafe):
        super().__init__(cafe)
        
    def get_custo(self):
        custo = super().get_custo()
        custo = custo + 1.00
        return custo
    
    def get_descricao(self):
        descricao = super().get_descricao()
        descricao = f"{descricao} + Açucar"
        return descricao

File: CSharp/FaculdadeVincit.DesignPatterns.DecoratorPattern/CafeAdicionalDeAcucar.cs
namespace FaculdadeVincit.DesignPatterns.DecoratorPattern;

public class CafeAdicionalDeAcucar : CafeDecorator
{
    public CafeAdicionalDeAcucar(ICafe cafe) : base(cafe)
    {
    }

    public override string Descricao => $"{base.Descricao} + Açucar";

    public override double Custo => base.Custo + 1.00;
}
